practice-two/src/services/customerService.js

```diff
@@ -1,12 +1,58 @@
-// import { BASE_URL, PATH } from "@constants";
+import { BASE_URL, PATH } from '@constants';
 
+// Get customer list
 const getAllCustomerService = async (url) => {
   const response = await fetch(url);
   if (!response.ok) {
-    throw new Error("An error occurred while fetching the data.");
+    throw new Error('An error occurred while fetching the data.');
   }
   const data = await response.json();
   return data;
 };
 
-export default getAllCustomerService;
+const service = async (path, method, data) => {
+  const url = `${BASE_URL}/${path}`;
+  try {
+    const response = await fetch(url, {
+      method,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (response.ok) {
+      return { data: await response.json(), error: null };
+    } else {
+      const errorData = await response.json();
+      return { error: errorData };
+    }
+  } catch (error) {
+    return { error };
+  }
+};
+
+// Create customer
+const createCustomerService = async (customerData) => {
+  const response = await service(`${PATH}`, 'POST', customerData);
```

Comment:
service should services

Reply:
the same

Reply:
I updated the name
![image](https://github.com/datdragon777/React-Training/assets/74806826/5ea2e0f8-f5a5-4e99-8276-8fb7ddb960aa)